src/lib/spsp.js

```diff
@@ -36,19 +39,41 @@ const _querySPSP = function * (receiver) {
     .set('Accept', 'application/json')).body
 }
 
-const query = co.wrap(_querySPSP)
+const _createPayment = (spsp, quote, id) => {
+  return {
+    id: id || uuid(),
+    sourceAmount: quote.sourceAmount,
+    destinationAmount: quote.destinationAmount,
+    destinationAccount: spsp.destination_account,
+    connectorAccount: quote.connectorAccount,
+    sourceExpiryDuration: quote.sourceExpiryDuration,
+    spsp: spsp
+  }
+}
 
-const _quote = function * ({ plugin, spsp, sourceAmount, destinationAmount, ilp }) {
-  if (!plugin) throw new Error('missing plugin')
-  if (!spsp.destination_account) throw new Error('missing destination account')
-  if (!spsp.maximum_destination_amount) throw new Error('missing maximum destination amount')
-  if (!spsp.minimum_destination_amount) throw new Error('missing minimum destination amount')
+const query = co.wrap(_querySPSP)
 
-  const client = new IlpCore.Client(plugin, ilp)
-  const quote = yield client.quote({
+const quote = function * (plugin, {
```

Comment:
I think the actual SPSP server and the ILQP requests should be integer amounts. But the SPSP functions that the user calls could use decimal amounts.